pkg/controller/volume/selinuxwarning/selinux_warning_controller.go

```diff
@@ -187,6 +187,28 @@ func (c *Controller) enqueuePod(_ klog.Logger, obj interface{}) {
 	c.queue.Add(podRef)
 }
 
+func (c *Controller) updatePod(logger klog.Logger, oldObj, newObj interface{}) {
+	// Pod.Spec fields that are relevant to this controller are immutable after creation (i.e.
+	// pod volumes, SELinux labels, privileged flag). React to update only when the Pod
+	// reaches its final state - kubelet will unmount the Pod volumes and the controller should
+	// therefore remove them from the cache.
+	oldPod, ok := oldObj.(*v1.Pod)
+	if !ok {
+		return
+	}
+	newPod, ok := newObj.(*v1.Pod)
+	if !ok {
+		return
+	}
+	if oldPod.Status.Phase == newPod.Status.Phase {
+		return
+	}
+	if newPod.Status.Phase != v1.PodFailed && newPod.Status.Phase != v1.PodSucceeded {
+		return
```

Comment:
This is basically an optimization right? The completed pods will anyways be skipped during processing below?

Reply:
I added a comment + squashed it into the commit that introduces this check.